crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -947,4 +962,26 @@ mod tests {
         assert!(entity.get_by_id(invalid_component_id).is_none());
         assert!(entity.get_mut_by_id(invalid_component_id).is_none());
     }
+
+    #[test]
+    fn entity_mut_world_scope_panic() {
+        let mut world = World::new();
+
+        let mut entity = world.spawn_empty();
+        let id = entity.id();
+        let res = std::panic::catch_unwind(AssertUnwindSafe(|| {
```

Comment:
Do you need the `AssertUnwindSafe` by making the closure `move`?

Reply:
This would move the variable `entity` into the closure and make it inaccessible to the rest of the test.